library.rb

```diff
@@ -0,0 +1,110 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+
+class Library
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    puts 'List all books'
+    @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+  end
+
+  def list_people
+    puts 'List al people'
+    @people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    option = gets.chomp.to_i
+
+    case option
+    when 1
+      create_student
+    when 2
+      create_teacher
+    end
+  end
+
+  def create_student
+    print 'Student age: '
+    age = gets.chomp
+    print 'Student name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp.upcase
+    case permission
+    when 'Y'
+      student = Student.new(nil, age, name, parent_permission: true)
+      @people << student
+    when 'N'
+      student = Student.new(nil, age, name, parent_permission: false)
+      @people << student
+    else
+      puts 'Invalid input!'
+    end
+    puts 'Person Created Successfully!'
+  end
```

Comment:
- [ ] In the `create_student` method, I would advise you to follow the `SOLID` principle while refactoring the code. All codes that ask the users for inputs or validate the user's input should be put in a different **_method, class or module._** The `create_student` method should be solely responsible for creating the student. 👍 💻  

#### Hint: 
Codes like this that ask users for input should be put in a separate **_method, class method or module method:_**

```ruby
print 'Student age: '
age = gets.chomp
print 'Student name: '
name = gets.chomp
```

Codes like this that validate users' inputs should be put in a separate **_method, class method or module method:_**

```ruby
 case permission
 when 'Y'
 parent_permission = true
```

Codes like this that create a new instance of the Student class should be put in a separate **_method, class method or module method:_**

```ruby
student = Student.new(nil, age, name, parent_permission: true)
@people << student
```

### N.B: Kindly implement these changes for the following methods:
- `create_teacher`
- `create_book`
- `create_rental`

Reply:
@Gambit142 done